### PR TITLE
Update Q&A of data_structure, array_and_linkedlist

### PR DESCRIPTION
**Pull request by longsizhuo**
There are no valuable questions worth recording in the first two chapters.

I'm not sure why there are 6 commits this time, because I actually only modified one file. I'm not sure why this phenomenon happened (maybe due to inexperience in using git).

Sry about that.

### CODE REVIEW

**Comment by longsizhuo** [comment]
I'd added some comments in the file.

File: docs/chapter_data_structure/summary.md [hunk 1]
```diff
@@ -17,8 +17,25 @@
 
 !!! question "为什么哈希表同时包含线性数据结构和非线性数据结构？"
 
-    哈希表底层是数组，而为了解决哈希冲突，我们可能会使用“链式地址”（后续哈希表章节会讲）。在拉链法中，数组中每个地址（桶）指向一个链表；当这个链表长度超过一定阈值时，又可能被转化为树（通常为红黑树）。因此，哈希表可能同时包含线性（数组、链表）和非线性（树）数据结构。
+    哈希表底层是数组，而为了解决哈希冲突，我们可能会使用“链式地址”（后续哈希表章节会讲）。在拉链法中，数组中每个地址（桶）指向一个链表；当这个链表长度超过一定阈值时，又可能被转化为树（通常为红黑树）。
+    从存储的角度来看，哈希表的底层是线性的数组结构。这个数组中的每一个槽位可能包含一个值，也可能包含一个链表或树（在解决哈希冲突时），因此，哈希表可能同时包含线性（数组、链表）和非线性（树）数据结构。
```
Comment (longsizhuo) [inline docs/chapter_data_structure/summary.md:21]: 有一个Question和这个类似，但是您的答案却有细微的不同，于是我在这里将两种回答都放在这里了，并且进行了一些语意调整。

File: docs/chapter_data_structure/summary.md [hunk 2]
```diff
@@ -17,8 +17,25 @@
...
     `char` 类型的长度由编程语言采用的编码方法决定。例如，Java、JS、TS、C# 都采用 UTF-16 编码（保存 Unicode 码点），因此 char 类型的长度为 2 bytes 。
+
+!!! question "基于数组实现的数据结构也被称为“静态数据结构” 是否有歧义？因为栈，队列也是可以进行出栈、入栈等操作的"
+
+    栈和队列确实可以实现动态的数据操作，但数据结构仍然是“静态”（长度不可变）的。尽管基于数组的数据结构可以动态地添加或删除元素，但它们的容量是固定的。如果数据量超出了预分配的大小，就需要创建一个新的更大的数组，并将老数组的内容复制到新数组中。
+
+!!! question "在构建栈/队列的时候，未指定栈/队列的大小，怎么理解成“静态数据结构呢”"
+
+    在高级编程语言中，我们无须人工指定栈/队列的初始容量，这个工作是由类内部自动完成的。例如，Java 的 ArrayList 的初始容量通常为 10 。另外，扩容操作也是自动实现的。详见本书的“列表”章节。
+
```
Comment (longsizhuo) [inline docs/chapter_data_structure/summary.md:34]: 此回答是上一个回答的追问，我觉得问得很好，于是单独提出来了

**Review by krahets — APPROVED** [review]
Thanks for the PR!

**Pull request merged**